go/vt/vttablet/tabletserver/rules/rules.go

```diff
@@ -462,6 +476,9 @@ func (qr *Rule) GetAction(
 	bindVars map[string]*querypb.BindVariable,
 	marginComments sqlparser.MarginComments,
 ) Action {
+	if qr.isCancelledFunc != nil && qr.isCancelledFunc() {
```

Comment:
`isCancelledFunc` makes the `Rue` dynamic; it can change behavior: the rule may expire.